File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const n = prices.length;
    const answer = Array.from({ length: n }, (_, index) => n - index - 1);
    const stack = [];

    for (let i = 0; i < n; i++) {
        while (stack.length && prices[i] < prices[stack.slice(-1)[0]]) {
            const top = stack.pop();
            answer[top] = i - top;
        }
        stack.push(i);
    }
    
    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    n = len(prices)
    stack = []
    answer = list(range(n-1, -1, -1))
    
    for i in range(n):
        while stack and prices[i] < prices[stack[-1]]:
            top = stack.pop()
            answer[top] = i - top
        stack.append(i)
    
    return answer
